File: examples/python/snippets/document/analyze_multi_page.py
from scanbotsdk import *

def analyze_multi_page(file_path: str):
    configuration = DocumentQualityAnalyzerConfiguration(
        tile_size=300,
        detect_orientation=True,
        min_estimated_number_of_symbols_for_document=20,
    )
    # Configure other parameters as needed.

    extractor = MultiPageImageExtractor()
    analyser = DocumentQualityAnalyzer(configuration=configuration)

    with RandomAccessSource(file_path=file_path) as file_access_source:
        extraction_result: PageExtractionResult = extractor.run(source=file_access_source)

        for page_index, page in enumerate(extraction_result.pages):
            for image_index, extracted_image in enumerate(page.images):
                # Important: wrap each extracted image in a context manager.
                # This ensures memory is released immediately after processing,
                # instead of waiting for the whole extraction_result to be freed.
                with extracted_image.image:
                    analysis_result = analyser.run(image=extracted_image.image)
                    print(
                        f"Page {page_index + 1}, Image {image_index + 1} "
                        f"-> Found: {analysis_result.document_found}, "
                        f"Quality: {analysis_result.quality}"
                    )

File: examples/nodejs/src/snippets/document/analyze-multipage.ts
import * as ScanbotSDK from "scanbotsdk";
import { extractImages } from "../utils/utils";

export class AnalyzeMultiPageSnippet {
  public static async run(filePath: string): Promise<void> {
    const analyzeConfig = new ScanbotSDK.DocumentQualityAnalyzerConfiguration();
    analyzeConfig.processByTileConfiguration.tileSize = 300;
    analyzeConfig.detectOrientation = true;
    analyzeConfig.minEstimatedNumberOfSymbolsForDocument = 20;
    // configure other parameters as needed

    // Extract images from the multi-page file
    await using extractionResult = await extractImages(filePath);

    // `await using` ensures the analyser is properly disposed
    // when the scope ends, as it holds unmanaged resources.
    await using analyzer = await ScanbotSDK.DocumentQualityAnalyzer.create(analyzeConfig);
    const pages = extractionResult.pages;
    console.log("Pages in document: " + pages.length);

    for (let pageIndex = 0; pageIndex < pages.length; pageIndex++) {
      const images = pages[pageIndex]!.images;

      for (let imageIndex = 0; imageIndex < images.length; imageIndex++) {
        // NOTE: Using `await using` on ImageRef is optional, because images are also
        // released when their parent container is disposed. However, images are stored
        // compressed and only decompressed on first access. Disposing them early
        // prevents a large number of decompressed images from piling up in memory.
        await using extractedImage = images[imageIndex]?.image;
        if (extractedImage) {
          // `await using` ensures the result is properly disposed when the scope ends, as it holds unmanaged resources.
          await using result = await analyzer.run(extractedImage);
          console.log(
            `Page ${pageIndex + 1}, Image ${imageIndex + 1} -> Found: ${
              result.documentFound
            }, Quality: ${result.quality}`
          );
        }
      }
    }
  }
}
